src/Security/BasicAuth.php

```diff
@@ -85,4 +107,90 @@ public static function getCurrentApplicationToken()
 
 		return self::$current_token;
 	}
+
+	/**
+	 * Fetch the user id via the auth header information
+	 *
+	 * @param boolean $do_login Perform a login request if not logged in
+	 *
+	 * @return integer User ID
+	 */
+	private static function getUserIdByAuth(bool $do_login = true):int
+	{
+		$a = DI::app();
+		Session::set('allow_api', false);
+		self::$current_user_id = 0;
+
+		// workaround for HTTP-auth in CGI mode
+		if (!empty($_SERVER['REDIRECT_REMOTE_USER'])) {
+			$userpass = base64_decode(substr($_SERVER["REDIRECT_REMOTE_USER"], 6));
+			if (strlen($userpass)) {
+				list($name, $password) = explode(':', $userpass);
+				$_SERVER['PHP_AUTH_USER'] = $name;
+				$_SERVER['PHP_AUTH_PW'] = $password;
+			}
+		}
```

Comment:
What are we keeping this for?

Reply:
I copied everything from the old place (except the old OAuth handling), because I cannot tell if this is needed or not.

Reply:
I was wondering if it was a value specific to Friendica for some reason but it doesn't look like it.

Reply:
Yeah, I also had to look if this was still a thing.